dbms/src/Common/TiFlashMetrics.h

```diff
@@ -374,18 +374,24 @@ namespace DB
       F(type_cftiny_fetch, {{"type", "cftiny_fetch"}}))                                                                             \
     M(tiflash_fap_task_result,                                                                                                      \
       "",                                                                                                                           \
-      Counter,                                                                                                                      \
+      Gauge,                                                                                                                        \
```

Comment:
Why this is a `Gauge` instead of `Counter` while you only call `...Increment()` on these metrics?

Reply:
Because `Counter` may be reset to zero after restart, which could be a problem

Reply:
> Because `Counter` may be reset to zero after restart, which could be a problem.

Using `Gauge` can avoid the metrics being reset to zero after restart?  How does it do it?

Reply:
I don't know. In fact during my test, `Gauge` is reset to zero after restart.
However, what I mean is when `Counter` is reset to zero, it will restart from zero as an offset from before the restart. For example, if the last recorded value was 10, and then a restart happened, and then the Counter reads 2, then prom will calibrate this value to 10+2 = 12, which I think is not suitable in this case.

Reply:
It has been changed back to Counter